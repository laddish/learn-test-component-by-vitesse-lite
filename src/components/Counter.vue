<script setup lang="ts">
const { initial = 10 } = defineProps<{
  initial: number
}>()

const { count, inc, dec } = useCounter(initial)
</script>

<template>
  <div>
    {{ count }}
    <button class="inc" @click="inc()">
      +
    </button>
    <button class="dec" @click="dec()">
      -
    </button>
  </div>
</template>
