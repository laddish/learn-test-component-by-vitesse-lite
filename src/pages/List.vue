<!-- <script setup lang="ts">
const movies = ref(
  [
    { title: 'The Shawshank Redemption', year: 1994 },
    { title: 'The Godfather', year: 1972 },
    { title: 'The Godfather: Part II', year: 1974 },
    { title: 'The Dark Knight', year: 2008 },
    { title: '12 Angry Monkeys', year: 2010 },
  ],
)
</script> -->

<script lang="ts">
export default {
  data() {
    return {
      movies: [
        { title: 'The Shawshank Redemption', year: 1994 },
        { title: 'The Godfather', year: 1972 },
        { title: 'The Godfather: Part II', year: 1974 },
        { title: 'The Dark Knight', year: 2008 },
        { title: '12 Angry Monkeys', year: 2010 },
      ],
    }
  },
}
</script>

<template>
  <div>
    <ul>
      <ListItem
        v-for="(movie, index) in movies"
        :key="index"
        :movie="movie"
      />
    </ul>
  </div>
</template>

<style scoped>

</style>
