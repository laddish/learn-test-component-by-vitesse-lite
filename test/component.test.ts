import { mount, shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Counter from '~/components/Counter.vue'
import List from '~/pages/List.vue'

describe('Counter.vue', () => {
  it('should render', () => {
    const wrapper = mount(Counter, { propsData: { initial: 10 } })
    expect(wrapper.text()).toContain('10')
    // 创建一个 snapshot 文件
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be interactive', async () => {
    const wrapper = mount(Counter, { props: { initial: 0 } })
    expect(wrapper.text()).toContain('0')

    expect(wrapper.find('.inc').exists()).toBe(true)

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('1')
  })
})

describe('List.vue', () => {
  it('List component shallow', async () => {
    // console.log(mount(List).html())
    // console.log(shallowMount(List).html())
    const wrapper = mount(List)
    console.log(wrapper.vm.movies)
    const movies = wrapper.vm.movies
    await wrapper.setData({
      movies: [...movies, {
        title: 'test',
        year: '2020',
      }],
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
