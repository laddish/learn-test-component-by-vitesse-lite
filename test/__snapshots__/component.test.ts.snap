// Vitest Snapshot v1

exports[`Counter.vue > should render 1`] = `"<div>10 <button class=\\"inc\\"> + </button><button class=\\"dec\\"> - </button></div>"`;

exports[`List.vue > List component shallow 1`] = `
"<div>
  <ul>
    <div>The Shawshank Redemption</div>
    <div>The Godfather</div>
    <div>The Godfather: Part II</div>
    <div>The Dark Knight</div>
    <div>12 Angry Monkeys</div>
    <div>test</div>
  </ul>
</div>"
`;
